File: RAI/AnimalExample/python_implementation/Cat.py
from Animal import Animal

class Cat(Animal):
    def __init__(self, name, specie, color) -> None:
        super().__init__(name, specie)
        self.color = color

    def purr(self):
        print("El gato " + self.name + " hace PURRRRRRRRR!!!!\n")
    
    def make_sound(self):
        print("El gato " +self.name + " hace MIAUUU...\n")

    def to_string(self):
        return "El Gato " + self.name + " de color " + self.color

    def __str__(self):
        return "El Gato " + self.name + " de color " + self.color

File: RAI/AnimalExample/JavaImplementation/Cat.java
public class Cat extends Animal{
    private String pelaje;
    private String formaOjos;
    private int capacidadSalto;

    public Cat(String nombre, String especie, String raza, String pelaje, String formaOjos, int capacidadSalto){
        super(nombre,especie,raza);
        this.pelaje = pelaje;
        this.formaOjos = formaOjos;
        this.capacidadSalto = capacidadSalto;
    }

    public void ronrronea(){
        System.out.println("El gato " + this.name + "hace purrrrrrrrrrrrr...");
    }

    public void makeSound(){
        System.out.println("El gato " + this.name + " hace miauuuu...");
    }
}